src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -459,9 +470,9 @@ private void ReceivePoll(Internal.Poll<K, V> poll)
         private void Poll()
         {
             var currentAssignment = _consumer.Assignment.ToImmutableList();
-            var initialRebalanceInProcess = _rebalanceInProgress.Value;
+            var initialRebalanceInProcess = _rebalanceInProgress;
 
-            if (_requests.IsEmpty())
+            if (_rebalanceInProgress || _requests.IsEmpty())
```

Comment:
Fix, make sure we're not polling actual messages while we're in a rebalance state